src/library/scala/collection/immutable/Range.scala

```diff
@@ -153,19 +153,15 @@ extends scala.collection.AbstractSeq[Int]
   }
 
   @inline final override def foreach[@specialized(Unit) U](f: Int => U) {
-    validateMaxLength()
```

Comment:
It was, because it only comes up if you try to foreach a range that is longer than Int.MaxValue elements (throwing an exception rather than working).  But if someone wants to argue that this is a bad idea (I'd say it's good because it means that for (i <- Int.MinValue to Int.MaxValue) works instead of weirdly throwing an exception), I can find a way to include the appropriate test without too much expense.